ValveResourceFormat/Resource/ResourceTypes/KV3.cs

```diff
@@ -0,0 +1,590 @@
+            {
+                stateStack.Pop();
+                stateStack.Push(State.VALUE_NUMBER);
+                currentString.Clear();
+                currentString.Append(c);
+            }
+            //Flagged resource
+            else
+            {
+                stateStack.Pop();
+                stateStack.Push(State.VALUE_FLAGGED);
+                currentString.Clear();
+                currentString.Append(c);
+            }
+        }
+
+        //Reading a property name
+        private static void ReadPropName(char c)
+        {
+            //Stop once whitespace is encountered
+            if (whiteSpace.Contains(c))
+            {
+                stateStack.Pop();
+                stateStack.Push(State.SEEK_VALUE);
+                currentName = currentString.ToString();
+                return;
+            }
+
+            currentString.Append(c);
+        }
+
+        //Read a structure
+        private static void ReadValueStruct(char c)
+        {
+            //Ignore whitespace
+            if (whiteSpace.Contains(c))
+            {
+                return;
+            }
+
+            //Catch comments
+            if (c == '/')
+            {
+                stateStack.Push(State.COMMENT);
+                currentString.Clear();
+                currentString.Append(c);
+                return;
+            }
+
+            //Check for the end of the structure
+            if (c == '}')
+            {
+                KVObject value = objStack.Pop();
+                objStack.Peek().AddProperty(value.key, new KVValue(typeof(KVObject), value));
+                stateStack.Pop();
+                return;
+            }
+
+            //Start looking for the next property name
+            stateStack.Push(State.PROP_NAME);
+            currentString.Clear();
+            currentString.Append(c);
+        }
+        
+        //Read a string value
+        private static void ReadValueString(char c)
+        {
+            if (c == '"')
+            {
+                //Multiline string detected
+                if (currentString.ToString() == "\"")
+                {
+                    stateStack.Pop();
+                    stateStack.Push(State.VALUE_STRING_MULTI);
+                    currentString.Clear();
+                    return;
+                }
+                //String ending found
+                else if (currentString.Length > 0)
+                {
+                    stateStack.Pop();
+                    objStack.Peek().AddProperty(currentName, new KVValue(typeof(String), currentString.ToString()));
+                    return;
+                }
+            }
+
+            currentString.Append(c);
+        }
+
+        //Reading multiline string
+        private static void ReadValueStringMulti(char c)
+        {
+            //Check for ending
+            if (c == '"' && currentString.ToString().Substring(currentString.Length - 2) == "\"\"")
```

Comment:
Just a nitpick but valve docs state it also needs to have a `\n` there too.

[End string with a newline followed by three unescaped quotes (eg. the buffer looks like `\n\"\"\"`)](https://developer.valvesoftware.com/wiki/Dota_2_Workshop_Tools/KeyValues3#Multiline_String_Literals)